### ISSUE TITLE
Error after updating the Orleans version

### ISSUE THREAD

**Issue by YunTengGroup** [OrleansContrib/OrleansDashboard#213]
After updating Microsoft.Orleans.Server to 2.2.4
An exception occurred while calling UseDashboard
Message:Multiple custom attributes of the same type found

**Comment by alexmg** [comment]
Here is the exception and stack trace.

```
System.Reflection.AmbiguousMatchException: 'Multiple custom attributes of the same type found.'

   at System.Attribute.GetCustomAttribute(Assembly element, Type attributeType, Boolean inherit)
   at System.Reflection.CustomAttributeExtensions.GetCustomAttribute[T](Assembly element)
   at Orleans.CodeGenerator.RoslynCodeGenerator.ShouldGenerateCodeForAssembly(Assembly assembly) in D:\build\agent\_work\25\s\src\Orleans.CodeGeneration\RoslynCodeGenerator.cs:line 653
   at System.Linq.Enumerable.WhereListIterator`1.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Orleans.CodeGenerator.RoslynCodeGenerator.GenerateAndLoadForAssemblies(IEnumerable`1 assemblies) in D:\build\agent\_work\25\s\src\Orleans.CodeGeneration\RoslynCodeGenerator.cs:line 109
   at Orleans.Hosting.ApplicationPartManagerCodeGenExtensions.WithCodeGeneration(IApplicationPartManagerWithAssemblies manager, ILoggerFactory loggerFactory) in D:\build\agent\_work\25\s\src\Orleans.CodeGeneration\ApplicationPartManagerCodeGenExtensions.cs:line 46
   at Orleans.ServiceCollectionExtensions.<>c.<UseDashboard>b__0_0(IApplicationPartManager appParts)
   at Orleans.Hosting.SiloHostBuilderExtensions.ConfigureApplicationParts(ISiloHostBuilder builder, Action`1 configure) in D:\build\agent\_work\23\s\src\Orleans.Runtime.Abstractions\Hosting\Generic\SiloHostBuilderExtensions.cs:line 153
   at Orleans.ServiceCollectionExtensions.UseDashboard(ISiloHostBuilder builder, Action`1 configurator)
```

**Comment by topnotch48** [comment]
have the same issue in our project :(

**Comment by richorama** [comment]
Sounds like a serious problem. I’ll take a look tomorrow. PRs are welcome!

**Comment by richorama** [comment]
tracking issue https://github.com/dotnet/orleans/issues/5398

**Comment by ReubenBond** [comment]
Fix PR: https://github.com/dotnet/orleans/pull/5407

**Comment by richorama** [comment]
fixed by version 2.2.4
